File: Task-03/Java/subtask4.java
import java.io.File;
import java.io.FileWriter;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        // Read the number
        Scanner fileScanner = new Scanner(new File("input.txt"));
        int n = fileScanner.nextInt(); 
        fileScanner.close();
        
        StringBuilder output = new StringBuilder();

        // Upper half of the diamond
        for (int i = 1; i <= n; i += 2) {
            output.append(" ".repeat((n - i) / 2));
            output.append("*".repeat(i)).append("\n");
        }

        // Lower half of the diamond
        for (int i = n - 2; i >= 1; i -= 2) {
            output.append(" ".repeat((n - i) / 2));
            output.append("*".repeat(i)).append("\n");
        }

        // Write the diamond pattern to output.txt
        FileWriter fileWriter = new FileWriter("output.txt");
        fileWriter.write(output.toString());
        fileWriter.close();
    }
}

File: Task-03/Rust/subtask4.rs
use std::fs::File;
use std::io::{self, Read, Write};

fn main() -> io::Result<()> {
    // Step : Read the value of n from input.txt
    let mut input_file = File::open("input.txt")?;
    let mut input = String::new();
    input_file.read_to_string(&mut input)?;

    let n: usize = input.trim().parse().expect("Please enter a valid number");

    let mut output = String::new();

    // Generate the upper part of the diamond
    for i in 1..=n {
        if i % 2 == 1 { // Only for odd numbers
            let spaces = (n - i) / 2;
            let stars = i;
            output.push_str(&format!("{}{}\n", " ".repeat(spaces), "*".repeat(stars)));
        }
    }

    // Generate the lower part of the diamond
    for i in (1..n).rev() {
        if i % 2 == 1 { // Only for odd numbers
            let spaces = (n - i) / 2;
            let stars = i;
            output.push_str(&format!("{}{}\n", " ".repeat(spaces), "*".repeat(stars)));
        }
    }

    let mut output_file = File::create("output.txt")?;
    output_file.write_all(output.as_bytes())?;

    // Return Ok if everything is successful
    Ok(())
}
